source/PlayerInfo.cpp

```diff
@@ -2639,6 +2639,111 @@ void PlayerInfo::RegisterDerivedConditions()
+	auto &&totalShipsProvider = conditions.GetProviderNamed("total ships");
+	totalShipsProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed())
+				++retVal;
+		return retVal;
+	});
+
+	// Conditions to determine what outfits the player owns, with various possible locations to check.
+	// The following condition checks all sources of outfits which are present with the player.
+	// If in orbit, this means checking all ships in-system for installed and in cargo outfits.
+	// If landed, this means checking all landed ships for installed outfits, the pooled cargo hold,
+	// and any planetary storage.
+	auto &&presentOutfitProvider = conditions.GetProviderPrefixed("outfit (present): ");
+	presentOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		const Planet *planet = GetPlanet();
+		if(planet)
+		{
+			retVal += Cargo().Get(outfit);
+			if(planetaryStorage.count(planet))
+				retVal += planetaryStorage[planet].Get(outfit);
+		}
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(ship->IsDestroyed() || ship->GetSystem() != system)
+				continue;
+			// If not on a planet, parked ships don't count. If on a planet, disabled ships don't count.
+			if((!planet && !ship->IsParked()) || (planet && !ship->IsDisabled()))
+			{
+				retVal += ship->OutfitCount(outfit);
+				retVal += ship->Cargo().Get(outfit);
+			}
+		}
+		return retVal;
+	});
+
+	// The following condition checks the player's entire fleet for installed outfits.
+	auto &&fleetOutfitProvider = conditions.GetProviderPrefixed("outfit (fleet installed): ");
+	fleetOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		for(const shared_ptr<Ship> &ship : ships)
+		{
+			if(ship->IsDestroyed())
+				continue;
+			retVal += ship->OutfitCount(outfit);
+		}
+		return retVal;
+	});
+
+	// The following condition checks all cargo in the player's fleet.
+	auto &&fleetCargoOutfitProvider = conditions.GetProviderPrefixed("outfit (fleet cargo): ");
+	fleetCargoOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		if(GetPlanet())
+			retVal += Cargo().Get(outfit);
+		for(const shared_ptr<Ship> &ship : ships)
+			if(!ship->IsDestroyed() && !ship->IsParked())
+				retVal += ship->Cargo().Get(outfit);
+		return retVal;
+	});
+
+	// The following condition checks the flagship's installed outfits.
+	auto &&flagshipOutfitProvider = conditions.GetProviderPrefixed("outfit (flagship installed): ");
+	flagshipOutfitProvider.SetGetFunction([this](const string &name) -> int64_t
+	{
+		int64_t retVal = 0;
+		const Outfit *outfit = GameData::Outfits().Get(name);
+		if(flagship)
```

Comment:
This function can be shortened and optimized; I suggest to first check if the player has a flagship, and to immediately return 0 if not.
The temporary value is also not needed, since `flagship->OutfitCount(outift)` can be returned directly.